test/jsonapi/plugs/content_type_negotiation_test.exs

```diff
@@ -14,17 +14,39 @@ defmodule JSONAPI.ContentTypeNegotiationTest do
     assert 415 == conn.status
   end
 
-  test "halts and returns an error if accept header is incorrect" do
+  test "halts and returns an error if content-type header contains other media type params" do
+    conn =
+      :post
+      |> conn("/example", "application/vnd.api+json; charset=utf-8")
```

Comment:
This is just setting the post body isn't it?  Wouldn't it be clearer to do something like `Plug.Conn.put_req_header("content-type", "text/html")`?